src/test/java/io/jenkins/plugins/analysis/core/model/IssueDifferenceTest.java

```diff
@@ -0,0 +1,63 @@
+package io.jenkins.plugins.analysis.core.model;
+
+import org.junit.jupiter.api.Test;
+
+import edu.hm.hafner.analysis.Issue;
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.LineRange;
+import edu.hm.hafner.analysis.LineRangeList;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+
+import static java.util.Collections.*;
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Tests the class {@link IssueDifference}.
+ *
+ * @author Artem Polovyi
+ */
+class IssueDifferenceTest {
```

Comment:
I think there is also a bug in my algorithm. Can you please create such a test (it should fail). I'm not sure but it could be the reason for https://issues.jenkins-ci.org/browse/JENKINS-56324:

current:
- issue A: message A, fingerprint F
- issue B: message B, fingerprint F

reference:
- issue B: message B, fingerprint F

I think then the algorithm wrongly decides that issue A has been fixed and issue B is new

Reply:
Since all 3 issues have the same fingerprint, the algorithm decides that _Current Issue A_ equals _Reference Issue B_. And treats _Issue A_ as outstanding, because there is same _Issue B_ in reference. 

But if you change issue A fingerprint (E.g. "F1") then the algorithm classifies Issue B as  outstanding and Issue A as new, which is in my opinion correct. Though I'm not sure in the fingerprint purpose, please correct if I'm wrong.

Reply:
Yes, currently the algorithm decides that Issue A is outstanding. But that is wrong. Issue A is new and Issue B is outstanding.

Reply:
My point is that the algorithm decides that Issue A is outstanding only because it has the same fingerprint as reference issue B. Which wouldn't be the case, since current issue A and reference issue B are two different issues which can't have same fingerprint?

Reply:
I changed fingerprint of issue A to "F1" and it works correctly in test `shouldCreateIssueDifferenceWithNewAndOutstandingIssue`.

Or is it possible to have different issues with same fingerprint?